### DISCUSSION TITLE
Fail to install Autogen on M1 MAC (package architecture (amd64) does not match system (arm64)

### DISCUSSION THREAD

**Discussion by AJFourier** [microsoft/autogen#1843] [Q&A]
I'm trying to install Autogen on an M1 MAC MINI, but getting the error below:
Can someone explain me how to solve this/

21.26 2024-03-02 14:40:21 (50.1 MB/s) - ‘quarto-1.5.23-linux-amd64.deb’ saved [122169950/122169950]
21.26 
21.27 dpkg: error processing archive quarto-1.5.23-linux-amd64.deb (--install):
21.27  package architecture (amd64) does not match system (arm64)
21.28 Errors were encountered while processing:
21.28  quarto-1.5.23-linux-amd64.deb
------
.devcontainer/Dockerfile:12
--------------------
  11 |     ENV DEBIAN_FRONTEND=noninteractive
  12 | >>> RUN apt-get update \
  13 | >>>    && apt-get upgrade -y  \
  14 | >>>    && apt-get -y install --no-install-recommends build-essential npm \
  15 | >>>    && apt-get autoremove -y \
  16 | >>>    && apt-get clean -y \
  17 | >>>    && wget https://github.com/quarto-dev/quarto-cli/releases/download/v1.5.23/quarto-1.5.23-linux-amd64.deb \
  18 | >>>    && dpkg -i quarto-1.5.23-linux-amd64.deb \
  19 | >>>    && rm -rf /var/lib/apt/lists/* quarto-1.5.23-linux-amd64.deb
  20 |     ENV DEBIAN_FRONTEND=dialog
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update    && apt-get upgrade -y     && apt-get -y install --no-install-recommends build-essential npm    && apt-get autoremove -y    && apt-get clean -y    && wget https://github.com/quarto-dev/quarto-cli/releases/download/v1.5.23/quarto-1.5.23-linux-amd64.deb    && dpkg -i quarto-1.5.23-linux-amd64.deb    && rm -rf /var/lib/apt/lists/* quarto-1.5.23-linux-amd64.deb" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/desktop-linux/desktop-linux/tvdhb4m8bre76t7zauew3y0dy

**Comment by AJFourier** [comment]
any help would be really appreciated

**Reply by tfoutrein** [reply]
I have exactly the same on MAC m2 due to ARM architecture

**Reply by AJFourier** [reply]
@tfoutrein did you manage to install it in any other way?

**Reply by tfoutrein** [reply]
i 've just resolved the pb 
Create a .devcontainer folder with : 

- devcontainer.json file copied from autogen base devcontainer config (https://github.com/microsoft/autogen/blob/main/.devcontainer/devcontainer.json)

- copy Dockerfile from the same directory (https://github.com/microsoft/autogen/blob/main/.devcontainer/Dockerfile) 

then change **quarto-1.5.23-linux-amd64.deb** by **quarto-1.5.23-linux-arm64.deb** everywhere in then dockerfile

After that you can build it without any pb : 
`docker build -f ./Dockerfile -t autogen_base_img .`

**Reply by AJFourier** [reply]
Thanks! I will give it a try.

**Reply by jackgerrits** [reply]
I've submitted a fix for this issue here: https://github.com/microsoft/autogen/pull/1866

**Reply by AJFourier** [reply]
@jackgerrits thanks so much
